elasticsearch_dsl/interfaces.py

```diff
@@ -0,0 +1,3283 @@
+    :arg low_freq_operator: No documentation available.
+    :arg minimum_should_match: No documentation available.
+    :arg query: (required) No documentation available.
+    :arg boost: Floating point number used to decrease or increase the
+        relevance scores of the query. Boost values are relative to the
+        default value of 1.0. A boost value between 0 and 1.0 decreases
+        the relevance score. A value greater than 1.0 increases the
+        relevance score.
+    :arg _name: No documentation available.
+    """
+
+    analyzer: Union[str, "DefaultType"]
+    cutoff_frequency: Union[float, "DefaultType"]
+    high_freq_operator: Union[Literal["and", "or"], "DefaultType"]
+    low_freq_operator: Union[Literal["and", "or"], "DefaultType"]
+    minimum_should_match: Union[int, str, "DefaultType"]
+    query: Union[str, "DefaultType"]
+    boost: Union[float, "DefaultType"]
+    _name: Union[str, "DefaultType"]
+
+    def __init__(
+        self,
+        *,
+        analyzer: Union[str, "DefaultType"] = DEFAULT,
+        cutoff_frequency: Union[float, "DefaultType"] = DEFAULT,
+        high_freq_operator: Union[Literal["and", "or"], "DefaultType"] = DEFAULT,
+        low_freq_operator: Union[Literal["and", "or"], "DefaultType"] = DEFAULT,
+        minimum_should_match: Union[int, str, "DefaultType"] = DEFAULT,
+        query: Union[str, "DefaultType"] = DEFAULT,
+        boost: Union[float, "DefaultType"] = DEFAULT,
+        _name: Union[str, "DefaultType"] = DEFAULT,
+        **kwargs: Any,
+    ):
+        if analyzer != DEFAULT:
+            kwargs["analyzer"] = analyzer
+        if cutoff_frequency != DEFAULT:
+            kwargs["cutoff_frequency"] = cutoff_frequency
+        if high_freq_operator != DEFAULT:
+            kwargs["high_freq_operator"] = high_freq_operator
+        if low_freq_operator != DEFAULT:
+            kwargs["low_freq_operator"] = low_freq_operator
+        if minimum_should_match != DEFAULT:
+            kwargs["minimum_should_match"] = minimum_should_match
+        if query != DEFAULT:
+            kwargs["query"] = query
+        if boost != DEFAULT:
+            kwargs["boost"] = boost
+        if _name != DEFAULT:
+            kwargs["_name"] = _name
+        super().__init__(**kwargs)
+
+
+class CoordsGeoBounds(AttrDict[Any]):
+    """
+    :arg top: (required) No documentation available.
+    :arg bottom: (required) No documentation available.
+    :arg left: (required) No documentation available.
+    :arg right: (required) No documentation available.
+    """
+
+    top: Union[float, "DefaultType"]
+    bottom: Union[float, "DefaultType"]
+    left: Union[float, "DefaultType"]
+    right: Union[float, "DefaultType"]
+
+    def __init__(
+        self,
+        *,
+        top: Union[float, "DefaultType"] = DEFAULT,
+        bottom: Union[float, "DefaultType"] = DEFAULT,
+        left: Union[float, "DefaultType"] = DEFAULT,
+        right: Union[float, "DefaultType"] = DEFAULT,
```

Comment:
The reason I'm putting defaults on all attributes including required ones is that I don't have complete trust that the schema is right, so I'm allowing some flexibility to be tolerant of different schema versions. For example, we are using the 8.15 schema now, but if the user connects to an older stack then what's required or optional might be different and I don't want code to break because of schema differences. The idea is to enhance the IDE experience, not to enforce arguments. Does that make sense?

Reply:
I suppose it does, but the IDE is going to tell you when something is mandatory, right? Also, yes, the specification currently has many gaps, but it's usually missing attributes, not having too much attributes. (Except for serverless, where we may have less things.) Oh, but that's why you have **kwargs.

It was tempting to say "we generate code as precise as possible, and we allow to fall-back to using a body parameter". But maybe that does not make sense here.

Reply:
I'm open to discuss this, but what I decided to do is to leave the default value, but add the required flag to the docstring. For example:

```
    """
    :arg query: (required) Text, number, boolean value or date you wish to
        find in the provided field.
    """
```

I also intended to move the required fields to the top of the argument list, but now I see that this is not working correctly, so I'll fix that.